### PR TITLE
infoschema: update raw args to make it drop correct table in `RENAME TABLE`

### LINKED ISSUE #43714: deleted system table still can be get from information_schema
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

create table in mysql db
drop table
check with "select * from information_schema"

### 2. What did you expect to see? (Required)
can't find this table in information_schema
### 3. What did you see instead (Required)
still exist
### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #43714

Problem Summary:

Reproducible step:

```sql
create table test.t1 (id int primary key,a text);
insert test.t1 values(1,'334'),(4,'3443435'),(5,'fdf43t536653');
rename table test.t1 to mysql.t1;
SELECT * FROM information_schema.TABLES WHERE (TABLE_SCHEMA = 'mysql') AND (TABLE_NAME = 't1');
```

#43341 changes the behavior of `RENAME TABLE`: it takes two schema versions to complete. However, `applyDiff()` doesn't handle two `SchemaDiff`s properly. The same table is created twice in `infoschema`, causing the phantom described in #43714.

### What is changed and how it works?

During applying schema diff, we check if the table already exists before creating a new table. Thus, we can skip the second creation.

Other alternative fixes include

- Drop the table first and then create a new table(like what `CREATE TABLE` with foreign key does).
    - Change `job.Args` at second step, so that it contains the new schema ID.
    - There are risks if we change `job.Args`, it is hard to guarantee all the components that relying on this work correctly.
- Check if the old table is dropped. If it is not found, skip creating the second table.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by crazycs520 — APPROVED** [review]

**Review by zimulala — APPROVED** [review]
LGTM

File: ddl/table.go [hunk 1]
```diff
@@ -1177,10 +1178,17 @@ func finishJobRenameTable(d *ddlCtx, t *meta.Meta, job *model.Job) (int64, error
 		job.State = model.JobStateCancelled
 		return 0, errors.Trace(err)
 	}
+	oldRawArgs := job.RawArgs
```
Comment (zimulala) [inline ddl/table.go:1181]: Could  we add comments for this logic？

**Review by zimulala — APPROVED** [review]

**Review by crazycs520 — APPROVED** [review]

**Pull request merged**